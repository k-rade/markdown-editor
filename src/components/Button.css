.button {
    font-family: var(--font);
    font-size: 1rem;
    border: 1px solid transparent;
    outline: none;
    padding: 0.25rem 1rem;
    border-radius: 8px;
    font-weight: 600;
    color: var(--color-black);
    cursor: pointer;
}

.button:disabled {
    cursor: not-allowed;
}

.button-primary {
    background-color: var(--color-primary-500);
    border: 1px solid var(--color-primary-400);
    box-shadow: 0 2px 4px hsl(0, 0%, 0%);
}

.button-outline {
    color: var(--color-white);
    background-color: transparent;
    border: 1px solid var(--color-white);
}

.button-ghost {
    font-weight: 500;
    color: var(--color-white);
    background-color: transparent;
}

.button-ghost:hover {

    background-color: var(--color-gray-600);
}