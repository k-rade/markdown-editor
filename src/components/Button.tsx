interface Props {
  text: string;
  type?: "button" | "submit";
  onClick?: () => void;
  disabled?: boolean;
  variant?: "secondary" | "danger";
  className?: string;
}

const Button = ({
  text,
  type = "button",
  onClick,
  disabled,
  variant = "secondary",
  className,
}: Props) => {
  const getButtonStyle = (): string => {
    switch (variant) {
      case "danger":
        return "border-red-600 bg-red-500 text-white hover:bg-red-600 hover:border-red-700 dark:bg-red-700 dark:hover:bg-red-600 dark:hover:border-red-500";
      case "secondary":
        return "border-zinc-300 bg-zinc-50 hover:bg-zinc-100 text-zinc-800 dark:bg-zinc-700 dark:text-zinc-300 dark:border-zinc-600 dark:hover:bg-zinc-600 dark:hover:border-zinc-500";
      default:
        return "border-zinc-300 bg-zinc-50 hover:bg-zinc-100 text-zinc-800 dark:bg-zinc-700 dark:text-zinc-300 dark:border-zinc-600 dark:hover:bg-zinc-600 dark:hover:border-zinc-500";
    }
  };

  return (
    <button
      className={`${className} py-1.5 shadow-sm font-semibold text-sm border px-4 rounded-md dark:shadow-md ${getButtonStyle()}`}
      type={type}
      onClick={onClick}
      disabled={disabled}
    >
      {text}
    </button>
  );
};

export default Button;
