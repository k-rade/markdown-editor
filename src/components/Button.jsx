import "./Button.css";

const Button = ({ variant, onClick, label, disabled }) => {
  return (
    <button
      disabled={disabled}
      onClick={onClick}
      type="button"
      className={`button button-${variant}`}
    >
      {label}
    </button>
  );
};

export default Button;
