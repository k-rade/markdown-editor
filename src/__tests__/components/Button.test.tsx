import { fireEvent, render, RenderResult } from "@testing-library/react";
import Button from "../../components/Button";

describe("Button", (): void => {
  const mockOnClick = jest.fn();

  const renderTestComponent = (
    variant?: "secondary" | "danger"
  ): RenderResult =>
    render(<Button variant={variant} text="button" onClick={mockOnClick} />);

  it("should render as secondary by default", (): void => {
    const { baseElement } = renderTestComponent();

    expect(baseElement).toMatchSnapshot();
  });

  it("should render as danger button", (): void => {
    const { baseElement } = renderTestComponent("danger");

    expect(baseElement).toMatchSnapshot();
  });

  it("should call onClick handler", (): void => {
    const { getByText } = renderTestComponent();

    fireEvent.click(getByText("button"));

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
