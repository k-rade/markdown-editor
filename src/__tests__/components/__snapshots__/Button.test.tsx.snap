// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render as danger button 1`] = `
<body>
  <div>
    <button
      class="undefined py-1.5 shadow-sm font-semibold text-sm border px-4 rounded-md dark:shadow-md border-red-600 bg-red-500 text-white hover:bg-red-600 hover:border-red-700 dark:bg-red-700 dark:hover:bg-red-600 dark:hover:border-red-500"
      type="button"
    >
      button
    </button>
  </div>
</body>
`;

exports[`Button should render as secondary by default 1`] = `
<body>
  <div>
    <button
      class="undefined py-1.5 shadow-sm font-semibold text-sm border px-4 rounded-md dark:shadow-md border-zinc-300 bg-zinc-50 hover:bg-zinc-100 text-zinc-800 dark:bg-zinc-700 dark:text-zinc-300 dark:border-zinc-600 dark:hover:bg-zinc-600 dark:hover:border-zinc-500"
      type="button"
    >
      button
    </button>
  </div>
</body>
`;
